control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -130,21 +130,28 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, broker *eventing.Broker)
 
 	logger.Debug("Brokers and triggers config map updated")
 
+	// Update volume generation annotation of receiver pods
+	if err := r.UpdateReceiverPodsAnnotation(logger, brokersTriggers.VolumeGeneration); err != nil {
+		return statusConditionManager.failedToUpdateReceiverPodsAnnotation(err)
+	}
+
 	// Update volume generation annotation of dispatcher pods
 	if err := r.UpdateDispatcherPodsAnnotation(logger, brokersTriggers.VolumeGeneration); err != nil {
 		// Failing to update dispatcher pods annotation leads to config map refresh delayed by several seconds.
-		// For now, we use the configuration map only on the dispatcher side, which means we don't lose availability,
-		// since the receiver will accept events. So, log out the error and move on to the next step.
+		// Since the dispatcher side is the consumer side, we don't lose availability, and we can consider the Broker
```

Comment:
Why an update failure on receiver side is considered an hard error, while an update failure on dispatcher side is a soft error?

Reply:
After #37 we reject events to a non-existing Broker, which means that we cannot consider a Broker Ready if all receivers haven't got the Broker, so update failures to receiver pods is a hard failure.

On the other side, dispatcher pods care about Triggers, and the Broker object is used as a configuration prototype for all associated Triggers, so I thought that it's fine on the dispatcher side to receive eventually the update even if here eventually means seconds or minutes after the actual update. 

I'm open to suggestions and discussions about this point because there are implications with both approaches.

Reply:
That makes sense for now, i would prefer if you can dump this "difference" in the comments. I see you already pointed out why it's not a problem the failure of dispatcher annotation update, but you didn't explained this difference

Reply:
Sure.

Reply:
https://github.com/knative-sandbox/eventing-kafka-broker/pull/38/commits/f962da910445114c3d2be2e08a0b62046889bfa9